horovod/spark/common/store.py

```diff
@@ -181,7 +181,8 @@ def __init__(self, prefix_path, train_path=None, val_path=None, test_path=None,
         super().__init__()
 
     def exists(self, path):
-        return self.fs.exists(self.get_localized_path(path)) or self.fs.isdir(path)
+        localized_path = self.get_localized_path(path)
+        return self.fs.exists(localized_path) or self.fs.isdir(localized_path)
```

Comment:
The semantics of `fs.isdir` looks like it includes `fs.exists`, so the ` or self.fs.isdir(localized_path)` looks redundant. Can you please specify in a comment why this bit is needed? Otherwise, someone might remove it in the future.

Reply:
I am not sure why it includes `or self.fs.isdir(localized_path)` here, it is in `AbstractFilesystemStore`, maybe other filesystem (such as HDFS) require this ? 
For DBFSLocal and local fs, `fs.isdir` is redundant.

Reply:
I would like to keep calling `fs.isdir` in `AbstractFilesystemStore.exists` and let author of `AbstractFilesystemStore` to update it.

Reply:
You are right, I did not see that this existed before. It has been introduced here: https://github.com/horovod/horovod/commit/7a697111eef7d88899551c176e31cde5ab61545c#diff-68621af6dea1527be13ddb5823a2d54a48a3f7b8fa98dd17cfa6ec1e3d677fccR180

However, you are changing the argument of calling into `self.isdir` from `path` to `localized_path`.

@tgaddair do you remember why you introduced the seemingly redundand `isdir(path)` here?

@WeichenXu123 why do you think `AbstractFilesystemStore` is the right place for `exists` to use the localized path rather than the given path? This affects **all** FilesystemStore implementations

Reply:
OK I can make `DBFSLocalStore` override "exists" method.